### ISSUE TITLE
Chats overwritten for items with same sender

### ISSUE THREAD

**Issue by Archit404Error** [cuappdev/resell-react-native#98]
Instead of creating two separate chats for two items with the same seller, the chat for the first item is simply overwritten. This doesn't make much sense since it makes negotiating for two separate items hard -- even though it is a bit strange to have 2 separate chats with the same person. It seems like this functionality should be revisited -- if we do decide to keep one chat per seller then when I message a seller for a particular item the item should be included in my initial chat message, and also the negotiate button would have to allow me to select between multiple items to negotiate for

https://github.com/cuappdev/resell-react-native/assets/22649594/8d2c3bb4-dba1-4930-a5a1-cf4d894b5161

If we decide to go with one chat per item, we might have to change the backend model and routes (I'm not sure how this is implemented)

**Comment by zachseidner1** [comment]
Do you think it is ok to just keep 1 chat per user? Otherwise we would have to refactor how all the chats are being stored on Firebase and basically the way the entire chat system works. We could probably get the negotiation modal to show multiple products if necessary. Tbh I don't really even see the point of the negotiation modal bc all it does is it just places the text "The lowest I can accept for this item is $__" in the chat box based on what you press.

**Comment by Archit404Error** [comment]
Yeah you can keep 1 chat per user but then there'll probably have to be some extra designs where when you message a user for a specific item, the item is "mentioned" in your message -- I'm envisioning this being similar to how sharing a post with someone on instagram works. You'll probably have to also remove the gray text next to the user's name on the chat page indicating the item that's being discussed

So in other words if you go with the 1 chat approach this will require extra design work but little dev work, whereas if you go with the multiple chat approach this will require extra dev work but little design work. Since you guys have a lot of work right now we can probably prioritize the 1 chat approach to reduce dev work -- just make sure you communicate this with the pod at work session today!
